File: dotnet/net5/Watches-V5/interfaces.cs
using System;
using System.Collections.Generic;

namespace WatchesV5
{
    public interface IDisplayStrategy
    {
        void Display(DateTime time);
    }
    
    public interface IAlarmStrategy
    {
        void AddAlarm(Alarm alarm);
        IEnumerable<Alarm> GetAlarms(DateTime? fromTime = null, DateTime? toTime = null);
        void RemoveAlarm(Alarm alarm);
        Alarm FindAlarmByName(string alarmName);
    }
}

File: python/interfaces.py
from abc import ABC, abstractmethod


class Display_Behavior(ABC):
    @abstractmethod
    def display(self, time):
        pass

class Alarm_Behavior(ABC):
    @abstractmethod
    def add_alarm(self, alarm):
        pass

    @abstractmethod
    def get_alarms(self, from_time=None, to_time=None):
        pass

    @abstractmethod
    def remove_alarm(self, alarm):
        pass

    @abstractmethod
    def find_alarm_by_name(self, alarm_name):
        pass
